iup/__init__.py

```diff
@@ -152,153 +153,6 @@ def to_incident(self, group_cols: tuple[str,] | None) -> IncidentUptakeData:
-
-    if filters is not None:
-        frame = apply_filters(frame, filters)
-
-    frame = select_columns(frame, estimate_col, date_col, group_cols, date_format)
-
-    frame = insert_rollout(frame, rollout, group_cols)
-
-    frame = CumulativeUptakeData(frame)
-
-    return frame
-
-
-def fetch_nis(path: str) -> pl.DataFrame:
-    """
-    Get NIS data from its file path or address.
-
-    Parameters
-    path: str
-        file path or url address for NIS data to import
-
-    Returns
-    pl.DataFrame
-        NIS cumulative uptake data exactly as in its source
-    """
-    return pl.read_csv(path)
-
-
-def apply_filters(frame: pl.DataFrame, filters: dict | None) -> pl.DataFrame:
-    """
-    Apply filters to NIS data to remove unnecessary rows.
-
-    Parameters
-    frame: pl.DataFrame
-        NIS data in the midst of parsing
-    filters: dict | None
-        filters to remove unnecessary rows from the NIS data
-        keys are NIS column names and values are entries for rows to keep
-
-    Returns
-    pl.DataFrame
-        NIS cumulative uptake data with unnecessary rows removed
-
-    Details
-        If multiple entries in a column indicate that a row should be kept,
-        the dictionary value for that column may be a list of these entries.
-    """
-    if filters is not None:
-        filter_expr = pl.lit(True)
-        for k, v in filters.items():
-            filter_expr &= pl.col(k).is_in(pl.lit(v))
-        frame = frame.filter(filter_expr)
-
-    return frame
-
-
-def select_columns(
-    frame: pl.DataFrame,
-    estimate_col: str,
-    date_col: str,
-    group_cols: dict | None,
-    date_format: str,
-) -> pl.DataFrame:
-    """
-    Select the date, uptake, and any grouping columns of NIS uptake data.
-
-    Parameters
-    frame: pl.DataFrame
-        NIS data in the midst of parsing
-    estimate_col: str
-        name of the NIS column for the uptake estimate (population %)
-    date_col: str
-        name of the NIS column for the date
-    group_cols: dict | None
-        dictionary of the NIS columns for the grouping factors
-        keys are the NIS column names and values are the desired column names
-    date_format: str
-        format of the dates in the NIS date column
-
-    Returns
-        NIS cumulative uptake data with only the necessary columns
-
-    Details
-        Only the estimate, date, and grouping factor columns are kept.
-    """
-
-    if group_cols is not None:
-        frame = frame.select(
-            [estimate_col] + [date_col] + [pl.col(k) for k in group_cols.keys()]
-        ).rename(group_cols)
-    else:
-        frame = frame.select([estimate_col] + [date_col])
-
-    frame = (
-        frame.with_columns(
-            estimate=pl.col(estimate_col).cast(pl.Float64, strict=False),
-            date=pl.col(date_col).str.to_date(date_format),
-        )
-        .drop_nulls(subset=["estimate"])
-        .sort("date")
-    )
-
-    return frame
-
-
 def insert_rollout(
```

Comment:
`insert_rollout` had been called from inside `parse_nis` which is now gone. So `insert_rollout`'s job must be done somewhere else now. Maybe as another class method inside LinearIncidentUptakeModel?